structuralcodes/codes/ec2_2023/_section_8_1_bending.py

```diff
@@ -0,0 +1,593 @@
+        kconf_s (float): Effectiveness factor for the spacing of the
+            confinement reinforcement (non-dimensional).
+        delta_fcd (float): The increase in concrete design strength due to
+            confinement in MPa.
+
+    Returns:
+        float: The average concrete strength increase in MPa.
+    """
+    if fcd < 0:
+        raise ValueError(f'fcd must be non-negative. Got {fcd} instead.')
+    if kconf_b < 0:
+        raise ValueError(
+            f'kconf_b must be non-negative. Got {kconf_b} instead.'
+        )
+    if kconf_s < 0:
+        raise ValueError(
+            f'kconf_s must be non-negative. Got {kconf_s} instead.'
+        )
+    if delta_fcd < 0:
+        raise ValueError(
+            f'delta_fcd must be non-negative. Got {delta_fcd} instead.'
+        )
+
+    return fcd + kconf_b * kconf_s * delta_fcd
+
+
+def kconf_b_square_single(bcs: float, b: float) -> float:
+    """Calculate the kconf_b effectiveness factor for square members in
+    compression with single confinement reinforcement.
+
+    EN1992-1-1:2023 Table (8.1).
+
+    Args:
+        bcs (float): Width of the confined section in mm.
+        b (float): Width of the unconfined section in mm.
+
+    Returns:
+        float: The effectiveness factor kconf_b (non-dimensional).
+    """
+    if bcs < 0:
+        raise ValueError(f'bcs must be non-negative. Got {bcs} instead.')
+    if b < 0:
+        raise ValueError(f'b must be non-negative. Got {b} instead.')
+
+    return (1 / 3) * (bcs / b) ** 2
+
+
+def kconf_s_square_single(s: float, bcs: float) -> float:
+    """Calculate the kconf_s effectiveness factor for square members in
+    compression with single confinement reinforcement.
+
+    EN1992-1-1:2023 Table (8.1).
+
+    Args:
+        s (float): Spacing of the confinement reinforcement in mm.
+        bcs (float): Width of the confined section in mm.
+
+    Returns:
+        float: The effectiveness factor kconf_s (non-dimensional).
+    """
+    if s < 0:
+        raise ValueError(f's must be non-negative. Got {s} instead.')
+    if bcs < 0:
+        raise ValueError(f'bcs must be non-negative. Got {bcs} instead.')
+
+    return (max(1 - s / (2 * bcs), 0)) ** 2
+
+
+def kconf_b_circular(bcs: float, b: float) -> float:
+    """Calculate the kconf_b effectiveness factor for circular members in
+    compression with circular confinement reinforcement.
+
+    EN1992-1-1:2023 Table (8.1).
+
+    Args:
+        bcs (float): Diameter of the confined section in mm.
+        b (float): Diameter of the unconfined section in mm.
+
+    Returns:
+        float: The effectiveness factor kconf_b (non-dimensional).
+    """
+    if bcs < 0:
+        raise ValueError(f'bcs must be non-negative. Got {bcs} instead.')
+    if b < 0:
+        raise ValueError(f'b must be non-negative. Got {b} instead.')
+
+    return (bcs / b) ** 2
+
+
+def kconf_b_multiple(
+    bcsx: float, bcsy: float, b_i: List[float], bx: float, by: float
```

Comment:
Can b_i be ArrayLike as commented before? This would also increase the performance of the function avoiding explicit for loops